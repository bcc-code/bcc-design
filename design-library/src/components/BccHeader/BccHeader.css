@layer components {
  .bcc-header-overline {
    @apply text-caption text-secondary;
  }
  .bcc-header-underline {
    @apply text-caption text-secondary;
  }
  .bcc-header-title-wrapper {
    @apply flex items-center justify-between;
  }
  .bcc-header-title {
    @apply text-heading mb-0.5 truncate font-bold;
  }
}
