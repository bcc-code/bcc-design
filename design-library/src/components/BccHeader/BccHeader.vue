<script setup lang="ts">
type Props = {
  overline?: string;
  title?: string;
  underline?: string;
};

withDefaults(defineProps<Props>(), {});
</script>

<template>
  <div>
    <p class="text-caption text-secondary">{{ overline }}</p>
    <div class="flex items-center justify-between">
      <h2 class="text-heading mb-0.5 truncate font-bold">{{ title }}</h2>
      <slot name="infoRight" />
    </div>
    <p class="text-caption text-secondary">{{ underline }}</p>
  </div>
</template>
