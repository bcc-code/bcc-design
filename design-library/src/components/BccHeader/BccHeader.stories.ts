import BccHeader from "./BccHeader.vue";

import type { Meta, StoryFn } from "@storybook/vue3";

export default {
  title: "Components/BccHeader",
  component: BccHeader,
  argTypes: {

  },
} as Meta<typeof BccHeader>;

const Template: StoryFn<typeof BccHeader> = (args) => ({
  components: { BccHeader },
  setup() {
    return { args };
  },
  template: `
    <BccHeader v-bind="args" />
  `,
});

export const Example = Template.bind({});
Example.args = {

};
