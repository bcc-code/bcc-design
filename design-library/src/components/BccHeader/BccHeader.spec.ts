import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import BccHeader from "./BccHeader.vue";

describe("BccHeader", () => {
  it("renders title", async () => {
    const wrapper = mount(BccHeader, { props: { title: "Title" } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders all text", async () => {
    const wrapper = mount(BccHeader, {
      props: { title: "Title", overline: "Overline", underline: "Underline" },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
