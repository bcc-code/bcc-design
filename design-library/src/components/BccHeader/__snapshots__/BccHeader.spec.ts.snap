// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BccHeader > renders all text 1`] = `
"<div>
  <p class=\\"text-caption text-secondary\\">Overline</p>
  <div class=\\"flex items-center justify-between\\">
    <h2 class=\\"text-heading mb-0.5 truncate font-bold\\">Title</h2>
  </div>
  <p class=\\"text-caption text-secondary\\">Underline</p>
</div>"
`;

exports[`BccHeader > renders title 1`] = `
"<div>
  <p class=\\"text-caption text-secondary\\"></p>
  <div class=\\"flex items-center justify-between\\">
    <h2 class=\\"text-heading mb-0.5 truncate font-bold\\">Title</h2>
  </div>
  <p class=\\"text-caption text-secondary\\"></p>
</div>"
`;
